### PR TITLE
Fix broken integration tests

### PR DESCRIPTION
**Pull request by pbondoer**
With my last PR #12 I broke integration tests by mistake. Turns out the issue was a bit annoying to debug, as the way we use the library is not what would be expected from most websites.

I've restored the old behavior of starting tours where they left off (except when they were left at the last step, they are then restarted). I also adapted the integration tests to work.

Please let me know if you have any questions. Code review appreciated 🎉

### CODE REVIEW

File: application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java [hunk 1]
```diff
@@ -196,39 +196,41 @@ private void tourBoundToPage() throws Exception
         assertFalse(secondPage.isTourDisplayed());
         assertTrue(secondPage.hasResumeButton());
 
-        // Resume (to step 4)
+        // Resume (to step 1, as we have ended the tour)
```
Comment (vmassol) [inline application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java]: I'm assuming that you've changed the behavior slightly:
* If the tour is ended, it goes back to step 1 when restarted
* If the tour is closed, it resumes back to where it was left when restarted

Sounds ok to me.

File: application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java [hunk 2]
```diff
@@ -196,39 +196,41 @@ private void tourBoundToPage() throws Exception
...
         secondPage.close();
         assertFalse(secondPage.isTourDisplayed());
         assertTrue(secondPage.hasResumeButton());
 
-        // Go to an other page and then go back
-        TourHomePage.gotoPage();
-        secondPage = PageWithTour.gotoPage("TourCode", "TourClass");
-        assertFalse(secondPage.isTourDisplayed());
-        assertTrue(secondPage.hasResumeButton());
-
```
Comment (vmassol) [inline application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java]: Isn't this removing a test? Namely that we verify that if we go to another page that had a step in the tour, we have the resume button available and the tour can be resumed from that other page.

File: application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java (hunk 1 above)
Comment (pbondoer) [inline application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java]: https://jira.xwiki.org/browse/TOUR-53

File: application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java (hunk 2 above)
Comment (pbondoer) [inline application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java]: I've addressed this in https://github.com/xwiki-contrib/application-tour/pull/13/commits/0e759a61472d094d693bf66aaa49e40887a4308f

**Pull request merged**